File: Top 100 Codes/04-Important Codes related to Arrays/24-Finding Minimum sum of absolute difference of given array/FindingMinimumsumofabsolutedifferenceofgivenarray.py
def sum_of_min_abs_diff(arr):
    n = len(arr)
    if n < 2:
        return 0  # If there's only one element, no difference exists.

    arr.sort()  # Sorting the array to bring closer elements together
    total_sum = 0

    for i in range(n):
        if i > 0:
            left_diff = abs(arr[i] - arr[i - 1])
        else:
            left_diff = float('inf')

        if i < n - 1:
            right_diff = abs(arr[i] - arr[i + 1])
        else:
            right_diff = float('inf')

        if left_diff < right_diff:
            min_diff = left_diff
        else:
            min_diff = right_diff

        total_sum += min_diff

    return total_sum

# Example usage
arr = [4, 2, 1, 8, 5]
print("Sum of minimum absolute differences:", sum_of_min_abs_diff(arr))

File: Top 100 Codes/04-Important Codes related to Arrays/24-Finding Minimum sum of absolute difference of given array/FindingMinimumsumofabsolutedifferenceofgivenarray.cpp
#include <iostream>
#include <algorithm>
#include <climits>

using namespace std;

int sumOfMinAbsDiff(int arr[], int n) {
    if (n < 2) return 0; // If there's only one element, no difference exists.

    sort(arr, arr + n); // Sorting the array to bring closer elements together
    int sum = 0;

    for (int i = 0; i < n; i++) {
        int leftDiff, rightDiff;

        if (i > 0) {
            leftDiff = abs(arr[i] - arr[i - 1]);
        } else {
            leftDiff = INT_MAX;
        }

        if (i < n - 1) {
            rightDiff = abs(arr[i] - arr[i + 1]);
        } else {
            rightDiff = INT_MAX;
        }

        if (leftDiff < rightDiff) {
            sum += leftDiff;
        } else {
            sum += rightDiff;
        }
    }

    return sum;
}

int main() {
    int arr[] = {4, 2, 1, 8, 5};
    int n = sizeof(arr) / sizeof(arr[0]);

    cout << "Sum of minimum absolute differences: " << sumOfMinAbsDiff(arr, n) << endl;

    return 0;
}
